### DISCUSSION TITLE
Events extending

### DISCUSSION THREAD

**Discussion by andim27** [nylo-core/nylo#63] [General]
Nylo has a Events pattern ability.But this is one-way sending data strategy.
How to organize Event Bus pattern in Nylo?
There are many external packages for this but I think it would be very useful to have eventBus build in the Nylo

**Comment by agordn52** [comment]
Hi @andim27,

I'm curious but can't see what benefit this would be for the framework.
What would be a good use case?

Thanks,

**Reply by andim27** [reply]
It's because now events in Nylo is a one way transfer systems.It good suit in small project but does not enough in large.How to communicate with many separated class(widgets)? if we need send an receive information in dynamic mode?Use Backpack?Use Streams? EvenBus will be a good chose in such cases. Maybe some Rx** ability?But it needs some NyState-Extended systems.
I encountered a case that the framework lacks two-way data transfer in dynamic mode for complex projects (not primitive ones).Let's think together

**Reply by agordn52** [reply]
I've managed to build a reactive `StatefulWidget`, that can be updated from other StatefulWidgets & pages. I'll continue with more tests and post an update here with my findings. The library I'm testing with at the moment is [event_bus_plus](https://pub.dev/packages/event_bus_plus).
Thanks for the suggestion.

**Reply by andim27** [reply]
`https://pub.dev/packages/atom_event_bus` is suitable too.I've used it too.All is ok

**Reply by agordn52** [reply]
Thanks, I'll check that one out too 👍

**Reply by agordn52** [reply]
@andim27,

I've had success with the [event_bus_plus](https://pub.dev/packages/event_bus_plus) library integrating the event bus pattern.

Now it's possible to keep the state of a `NyState` in sync and update that state from anywhere in your application.
I'll write up some new docs for the website and release a [YouTube](https://www.youtube.com/@nylo_dev) video on how you can start using it in [5.2.0](https://github.com/nylo-core/framework/releases/tag/v5.2.0)

**Reply by sizoune** [reply]
> @andim27,
> 
> I've had success with the [event_bus_plus](https://pub.dev/packages/event_bus_plus) library integrating the event bus pattern.
> 
> Now it's possible to keep the state of a `NyState` in sync and update that state from anywhere in your application. I'll write up some new docs for the website and release a [YouTube](https://www.youtube.com/@nylo_dev) video on how you can start using it in [5.2.0](https://github.com/nylo-core/framework/releases/tag/v5.2.0)

hi, im curious, are you already post the video about how to manage state in different widget (or page)?

**Reply by agordn52** [reply]
> > @andim27,
> > I've had success with the [event_bus_plus](https://pub.dev/packages/event_bus_plus) library integrating the event bus pattern.
> > Now it's possible to keep the state of a `NyState` in sync and update that state from anywhere in your application. I'll write up some new docs for the website and release a [YouTube](https://www.youtube.com/@nylo_dev) video on how you can start using it in [5.2.0](https://github.com/nylo-core/framework/releases/tag/v5.2.0)
> 
> hi, im curious, are you already post the video about how to manage state in different widget (or page)?

Hi @sizoune,

Yeah, the video's here https://www.youtube.com/watch?v=X5EVh1KooFk&pp=ygUdbnlsbyBmbHV0dGVyIHN0YXRlIG1hbmFnZW1lbnQ%3D  👍

**Reply by sizoune** [reply]
![image](https://github.com/nylo-core/nylo/assets/13961066/dfa91985-a0fa-4c09-951d-87ce655a76c5)
im curious, im trying to update state in my Favorite Page (left side of the image), so in the Detail Page (right side of the image), i have Favorite Button to save the data into Favorite (storage), and i read in docs, we can set stateName in the NyList grid. the problem is when i tap on the favorite button, it doesnt take any effect

**Reply by agordn52** [reply]
> ![image](https://private-user-images.githubusercontent.com/13961066/346136006-dfa91985-a0fa-4c09-951d-87ce655a76c5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxODY0OTYsIm5iZiI6MTcyMDE4NjE5NiwicGF0aCI6Ii8xMzk2MTA2Ni8zNDYxMzYwMDYtZGZhOTE5ODUtYTBmYS00YzA5LTk1MWQtODdjZTY1NWE3NmM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA1VDEzMjk1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyZjA5NWY4ZjYwYTllNzgyOTQyNWIzYzc5YTAxODk1NTRhZjY2N2Y5ZGI0ZDk3Y2FiMTQ5MzRkMjhmZDI3YjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.arqlFD-k0qz4-yW6WWkLdti8HLvR7bkpyakF7mbDbFw) im curious, im trying to update state in my Favorite Page (left side of the image), so in the Detail Page (right side of the image), i have Favorite Button to save the data into Favorite (storage), and i read in docs, we can set stateName in the NyList grid. the problem is when i tap on the favorite button, it doesnt take any effect

You'll need to use a `StateAction`, here are the docs on it https://nylo.dev/docs/5.20.0/ny-state#state-actions

Try using `StateAction.refreshPage('your_state');` instead
